src/shared/hooks/wallets.ts

```diff
@@ -270,3 +275,29 @@ export function useWalletChange() {
     isStaked,
   ])
 }
+
+export function useCachedWalletName() {
+  const [accountName, setAccountName] = useState<string | null>(null)
+  const walletAddress = useDappSelector(selectWalletAddress)
+
+  useEffect(() => {
+    const handleCachedNamesUpdate = () => {
+      if (!walletAddress) return
+
+      const cachedNames = localStorage.getItem(LOCAL_STORAGE_CACHED_NAMES)
+      if (!cachedNames) return
+
+      const parsedCachedNames: CachedNames = JSON.parse(cachedNames)
+      const { ens, uns } = parsedCachedNames[walletAddress]
+
+      if (ens) setAccountName(ens.name)
+      if (uns) setAccountName(uns.name)
+    }
+
+    window.addEventListener("storage", handleCachedNamesUpdate)
+
+    return () => window.removeEventListener("storage", handleCachedNamesUpdate)
```

Comment:
Feels like we are doing something weird trying to communicate between a util and a hook via events on the window. I feel like we are trying to achieve the same thing here as we would have if we just triggered the change on the redux slice?